test/unit/ui/marker.test.js

```diff
@@ -17,62 +17,201 @@ function createMap() {
 
 test('Marker', (t) => {
     t.test('constructor', (t) => {
-        const el = window.document.createElement('div');
+        const el = window.document.createElement(`mapboxgl-marker`);
         const marker = new Marker(el);
         t.ok(marker.getElement(), 'marker element is created');
         t.end();
     });
 
     t.test('default marker', (t) => {
-        const marker = new Marker();
-        t.ok(marker.getElement(), 'default marker is created');
-        t.ok(marker.getOffset().equals(new Point(0, -14)), 'default marker with no offset uses default marker offset');
+        const el = window.document.createElement(`mapboxgl-marker`);
+        const marker = new Marker(el);
+        t.ok(marker.getElement(this.options), 'default marker is created');
+        t.ok(marker.getAnchor(), `marker set with anchor to default`);
+        t.ok(marker.getOffset(), 'default marker with no offset uses default marker offset');
         t.end();
     });
 
     t.test('default marker with some options', (t) => {
```

Comment:
let's leave the original test in place, you can write a new test for the functionality you've added.

Reply:
will do 👍